src/Test.UnitTests.Sarif/Core/LocationTests.cs

```diff
@@ -85,5 +92,160 @@ public void Location_LogicalLocation_WhenSetToNull_RemovesLogicalLocationsArray(
+                shouldSerialize: false, reconstructedLocationId: -1);
+
+            jsonLocation = $"{{\"id\":{new BigInteger(int.MinValue) - 1}}}";
+            VerifyIdRoundTripFromJsonHelper(jsonLocation, deserializedId: new BigInteger(int.MinValue) - 1,
+                shouldSerialize: false, reconstructedLocationId: -1);
+
+            jsonLocation = "{\"id\":-2}";
+            VerifyIdRoundTripFromJsonHelper(jsonLocation, deserializedId: -2,
+                shouldSerialize: false, reconstructedLocationId: -1);
+
+            jsonLocation = "{\"id\":-1}";
+            VerifyIdRoundTripFromJsonHelper(jsonLocation, deserializedId: -1,
+                shouldSerialize: false, reconstructedLocationId: -1);
+
+            jsonLocation = "{\"id\":0}";
+            VerifyIdRoundTripFromJsonHelper(jsonLocation, deserializedId: 0,
+                shouldSerialize: true, reconstructedLocationId: 0);
+
+            jsonLocation = "{\"id\":1}";
+            VerifyIdRoundTripFromJsonHelper(jsonLocation, deserializedId: 1,
+                shouldSerialize: true, reconstructedLocationId: 1);
+
+            jsonLocation = "{\"id\":2}";
+            VerifyIdRoundTripFromJsonHelper(jsonLocation, deserializedId: 2,
+                shouldSerialize: true, reconstructedLocationId: 2);
+
+            jsonLocation = $"{{\"id\":{int.MaxValue}}}";
+            VerifyIdRoundTripFromJsonHelper(jsonLocation, deserializedId: int.MaxValue,
+                shouldSerialize: true, reconstructedLocationId: int.MaxValue);
+
+            jsonLocation = $"{{\"id\":{new BigInteger(int.MaxValue) + 1}}}";
+            VerifyIdRoundTripFromJsonHelper(jsonLocation, deserializedId: new BigInteger(int.MaxValue) + 1,
+                shouldSerialize: true, reconstructedLocationId: new BigInteger(int.MaxValue) + 1);
+
+            jsonLocation = $"{{\"id\":{long.MaxValue}}}";
+            VerifyIdRoundTripFromJsonHelper(jsonLocation, deserializedId: long.MaxValue,
+                shouldSerialize: true, reconstructedLocationId: long.MaxValue);
+
+            jsonLocation = $"{{\"id\":{new BigInteger(long.MaxValue) + 1}}}";
+            VerifyIdRoundTripFromJsonHelper(jsonLocation, deserializedId: new BigInteger(long.MaxValue) + 1,
+                shouldSerialize: true, reconstructedLocationId: new BigInteger(long.MaxValue) + 1);
+
+            jsonLocation = $"{{\"id\":{ulong.MaxValue}}}";
+            VerifyIdRoundTripFromJsonHelper(jsonLocation, deserializedId: ulong.MaxValue,
+                shouldSerialize: true, reconstructedLocationId: ulong.MaxValue);
+
+            jsonLocation = $"{{\"id\":{new BigInteger(ulong.MaxValue) + 1}}}";
+            VerifyIdRoundTripFromJsonHelper(jsonLocation, deserializedId: new BigInteger(ulong.MaxValue) + 1,
+                shouldSerialize: true, reconstructedLocationId: new BigInteger(ulong.MaxValue) + 1);
+        }
+
+        [Fact]
+        public void Location_VerifyAbleToDeserializeWithBigIntegerId()
```

Comment:
>public void Location_VerifyAbleToDeserializeWithBigIntegerId() [](http://example.com/codeflow?start=8&length=62)

added a new test for `JsonConvert.DeserializeObject<SarifLog>(content)`
from a SARIF file.